models/hash_validator.rb

```diff
@@ -0,0 +1,273 @@
+# Copyright © Mapotempo, 2021
+#
+# This file is part of Mapotempo.
+#
+# Mapotempo is free software. You can redistribute it and/or
+# modify since you respect the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# Mapotempo is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the Licenses for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Mapotempo. If not, see:
+# <http://www.gnu.org/licenses/agpl.html>
+#
+
+require './models/vrp'
+
+module Models
+  class HashValidator < Vrp
+    def initialize(hash, internal_call)
+      @hash = hash
+      @internal_call = internal_call
+    end
+
+    def check_hash
```

Comment:
Do we really need a class with initialiser and a function? We can have the initialiser do the work and be done with it.

Reply:
I felt it could be weird to 'initialize' a validator and in fact it has already validated.

But I agree that we are always going to call them consecutively.

Reply:
In fact, I ask myself if we need a class for this logic. Because it is a hash anyways (i.e., the hash has no methods coming from its class that we call). Even though I don't like concerns, we already have several of them, we can have this logic in a "concerned" function without any class. What do you think @fonsecadeline, wouldn't it be more simple like that ?

Reply:
Concern works for me !

Reply:
The only thing is that if we use concern then we need to call it with something like 
```vrp.check_consistency``` 
whereas we want to check consistency on _hash_, not on _vrp_. This might be confusing don't you think ? Am I missing something?

Reply:
Since vrp.create(hash) is also called with an hash, it doesn't shock me if vrp.check_consistency(hash) is called with an hash to do the vrp validation.